Exiled.CustomItems/API/Features/CustomCandy.cs

```diff
@@ -0,0 +1,200 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="CustomCandy.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.CustomItems.API.Features
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Exiled.API.Features;
+    using Exiled.API.Features.Items;
+    using Exiled.API.Features.Pickups;
+    using Exiled.API.Features.Spawn;
+    using Exiled.Events.EventArgs.Scp330;
+    using InventorySystem.Items.Usables.Scp330;
+    using MEC;
+    using UnityEngine;
+
+    using Scp330Pickup = Exiled.API.Features.Pickups.Scp330Pickup;
+
+    /// <summary>
+    /// THe custom candy base class.
+    /// </summary>
+    public abstract class CustomCandy : CustomItem
+    {
+        private static new readonly Dictionary<ushort, Dictionary<int, CustomCandy>> TrackedSerials = new();
+
+        private float weight = 100;
+
+        /// <inheritdoc/>
+        /// <exception cref="NotSupportedException">Throws when trying to set value to property.</exception>
+        public override ItemType Type
+        {
+            get => ItemType.SCP330;
+            set => throw new NotSupportedException("Cannot set ItemType for CustomCandy. To set candy use CandyType.");
+        }
+
+        /// <summary>
+        /// Gets or sets a chance that candy will become custom.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Throws when <see langword="value"/> is less that 0 or above 100.</exception>
+        public override float Weight
+        {
+            get => weight;
+            set
+            {
+                if (value is > 100 or < 0)
+                    throw new ArgumentOutOfRangeException(nameof(value), value, "Value must be above zero and less that 100.");
+
```

Comment:
Why ?

Reply:
It can equals to 0 but here weight is a chance to pick up a custom candy instead of normal

Reply:
actually you remove the possibility to make it greater than 100 there no need for that

Reply:
Use the `Clamp` fonction of unity this will avoid errors that would break the workflow and avoid config errors